nchelpers/__init__.py

```diff
@@ -1059,9 +1063,12 @@ def multi_year_bounds(time_bounds, scale):
         if hasattr(time_var, 'bounds'):
             scale = time_scale(time_var)
             time_bounds_name = time_var.bounds
-            time_bounds = self.variables[time_bounds_name]
-            if multi_year_bounds(time_bounds, scale):
-                return time_bounds_name
+            try:
+                time_bounds = self.variables[time_bounds_name]
+                if multi_year_bounds(time_bounds, scale):
+                    return time_bounds_name
+            except KeyError:
+                pass  # claimed time bounds variable does not exist
```

Comment:
Likewise

Reply:
Again, this is a succession of rules/heuristics, which if they succeed return a value, if not they fall through to the next rule/heuristic.

Reply:
And similarly a bit tricky ... KeyError raised if the claimed bounds variable doesn't exist (fall through). If not, test whether the existing variable's values look plausible. If not, fall through.